README.md

```diff
@@ -1,2 +1,131 @@
+**Responsive Web Design** has become a popular and more flexible alternative.
+
+**Authentication** - the process of verifying that you are who you say you are
+E.g. when you login to your email you use a password to authenticate your
+identity. Other methods for authentication include using an ID card or
+fingerprint/biometric method.
+
+**Authorization** - check that you are authorised (have permission) to perform a
+certain action or to see a given page. E.g. admin users are authorised to create
+new users on the website.
+
+**CSS Box Model** - the concept that in HTML all elements are contained within
+rectangles or boxes. Each box consists of: margins, borders, padding, and the
+actual content.
+
+**Customer relationship management (CRM)** - a CRM system allows businesses to
+manage business relationships and the data and information associated with them.
+E.g. using CRM to track holiday bookings for a holiday bookings site. The CRM
+would have information on the customer and the trip they’re planning/ have
+booked and any staff members or actions that may need taking to progress or
+confirm their booking.
+
+**Create, Read, Update, and Delete (CRUD)** - CRUD refers to the 4 basic
+operations you can perform on data in a relational database application. Create
+a new data entry, read or retrieve existing data from the database, update the
+value of an existing piece of data or delete an existing piece of data.
+
+**Cross-Origin Resource Sharing (CORS)** - is a specification that allows
+specified resources on a web page to be requested from another domain outside
+the domain from which the first resource was served. E.g. uploading images using
+different domains using Amazon S3. For more about origin policies see
+'Same-origin Policy'.
+
+**Domain Name** - the characters/words used to identify a website after `http://` or
+`www.`. The domain name consists of two parts. Take this example:
+`https://dwyl.com`, `dwyl` makes up the **second level domain (SLD)** which is
+then followed by `.com` which what is known as the **top level domain (TLD)** or
+**parent domain**. Other examples of top level domains are: `.org`, `.co.uk`
+`.io`. Domain names are the more memorable and user-friendly representation of a
+website's **IP address**. Domain names are registered under the **Domain Name
+System (DNS)**.
+
+**Domain Name System (DNS)** - a distributed global directory of website domain
+names (and other internet resources). The directory stores already registered
+domain names along with their corresponding IP addresses.
+
+**Hypertext Transfer Protocol (HTTP)** - the set of rules for transferring files
+(text, graphic images, sound, video, and other multimedia files) on the World
+Wide Web. When you enter http:// in your address bar in front of the domain, it
+tells the browser to connect over HTTP.
+
+**Internet Protocol (IP) Address** - a unique series of numbers used to identify a website.
```

Comment:
This definition is _correct_ for IPv4. https://en.wikipedia.org/wiki/IPv4 ✅ 
It could be _extended_ for IPv6 https://en.wikipedia.org/wiki/IPv6
which allows addresses to be specified in Hexadecimal https://en.wikipedia.org/wiki/Hexadecimal